frontend/apps/filemanager/filemanagershortcuts.lua

```diff
@@ -190,13 +190,24 @@ function FileManagerShortcuts:deleteFolderShortcut(item)
 end
 
 function FileManagerShortcuts:onShowFolderShortcutsDialog()
-    local fm_bookmark =  FileManagerShortcuts:new{
+    local fm_bookmark = self:new{
         title = _("Folder shortcuts"),
         show_parent = self.ui,
-        curr_path = self.ui.file_chooser.path,
+        curr_path = self.ui.file_chooser and self.ui.file_chooser.path or self.ui:getLastDirFile(),
         goFolder = function(folder)
             if folder ~= nil and lfs.attributes(folder, "mode") == "directory" then
-                self.ui.file_chooser:changeToPath(folder)
+                if self.ui.file_chooser then
+                    self.ui.file_chooser:changeToPath(folder)
+                else
+                    local FileManager = require("apps/filemanager/filemanager")
+
```

Comment:
May be just add a comment for that `else` that in that case, we are called from the reader - that's what I suppose it's for  (and remove that blank like)

Reply:
What's against the blank line? :-)

Reply:
What is it for? :)
(Feels like you forgot it - I use these to indicate some different logical/work/context block of code - but it feels like it's not the case there.)

Reply:
In this case a combination of legibility and the logical separation between defining the variables being used and doing stuff.